src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/OrganisationServiceImpl.java

```diff
@@ -325,5 +391,49 @@ public OrganisationsDetailResponse findByOrganisationStatus(OrganisationStatus s
     }
 
 
+    @Override
+    public ResponseEntity retrieveOrganisationsWithPageable(Pageable pageable) {
+        List<OrganisationStatus> statusList = new ArrayList<>();
+        statusList.add(OrganisationStatus.PENDING);
+        statusList.add(OrganisationStatus.ACTIVE);
+        Page<Organisation> organisationsPage = organisationRepository.findByStatusIn(
+                statusList, pageable);
+
+        OrganisationsDetailResponse organisationsDetailResponse =
+                new OrganisationsDetailResponse(retrieveActiveOrganisationDetails(organisationsPage.getContent()),
+                        true);
+
+        if (organisationsDetailResponse.getOrganisations().isEmpty()) {
+            log.info("No Organisations Retrieved...");
+            throw new EmptyResultDataAccessException(1);
+        } else {
+            log.info("Retrieving all organisations..." + organisationsDetailResponse.getOrganisations().size());
+            HttpHeaders headers = RefDataUtil.generateResponseEntityWithPaginationHeader(pageable, organisationsPage, null);
+            return ResponseEntity.status(200).headers(headers).body(organisationsDetailResponse);
+        }
+
+    }
+
+    @Override
+    public ResponseEntity findByOrganisationStatusWithPageable(OrganisationStatus status, Pageable pageable) {
```

Comment:
Please refactor this in order to make the code more readable and obey single-responsisbility. I would recommend injecting an helper server aka OrganisationHelperService and moving the code there into well-named public methods. This will also make it easier to test.